storage/store/inmemory/unsynchronized/transactions.go

```diff
@@ -0,0 +1,44 @@
+package unsynchronized
+
+import (
+	"sync"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+)
+
+type Transactions struct {
+	//TODO: we don't need a mutex here as we have a guarantee by design
+	// that we write data only once and it happens before the future reads.
+	// We decided to leave a mutex for some time during active development.
+	// It'll be removed later on.
+	lock  sync.RWMutex
+	store map[flow.Identifier]*flow.TransactionBody
+}
+
+func NewTransactions() *Transactions {
+	return &Transactions{
+		store: make(map[flow.Identifier]*flow.TransactionBody),
+	}
+}
+
+var _ storage.Transactions = (*Transactions)(nil)
```

Comment:
nit: move this above the functions so it's easier to find when looking at the type.